### ISSUE TITLE
Compatibility with playbuzz plugin and embed?

### ISSUE THREAD

**Issue by gabu69** [Automattic/facebook-instant-articles-wp#134]
Is there any future support for playbuzz plugin and embeds?

https://wordpress.org/plugins/playbuzz/
https://publishers.playbuzz.com/how_to/embed-facebook-instant-article/

At the moment it wont display anything and gotta unpublish them manually all the time

**Comment by diegoquinteiro** [comment]
@gabu69 can you give me an example of the plugin output markup?

**Comment by IevaUzk** [comment]
Hi, I'm battling same thing at the moment. 

Shortcode:  
[playbuzz-item url="{game-url}"]

Embed code: 
<script type="text/javascript" src="//cdn.playbuzz.com/widget/feed.js"></script>
<div class="pb_feed" data-embed-by="{user-id}" data-game="{game-url}" ></div>

Rendered:
<script type="text/javascript" src="//cdn.playbuzz.com/widget/feed.js"></script>
<div style="<...>" class="pb_feed pb_identifier_0 pb_feed_anim " data-provider="WordPress 4.3.1" data-tags="" data-game="{game-url}" data-game-info="false" data-comments="false" data-shares="false" data-recommend="false" data-links="" data-width="auto" data-height="auto" data-margin-top="0"><div style="<...>" class="pb_top_content_container pb_feed_anim"><div><iframe style="min-width: 100%; width: 1px; max-width: 640px; height: 7216px; border: medium none; margin: auto; display: table; max-height: 7216px ! important;" iniframe="false" scrolling="no" class="pb_feed_iframe" name="pb_feed_iframe" src="{game-url}" frameborder="0" height="100%"></iframe></div></div><div style="display: none;" class="pb_feed_placeholder_container"><div class="pb_feed_placeholder_inner"><div class="pb_feed_placeholder_content"></div><div class="pb_feed_placeholder_preloader"><div class="pb_feed_calculator"><div class="pb_feed_calcwrapper"><div class="pb_feed_calc_top"></div><div class="pb_feed_calc_bottom"></div></div></div><div class="pb_feed_loading_text">LOADING</div></div></div></div></div>
	
<div class="don-share don-bubble-right" data-url="{game-url}" data-image="" data-style="icons" data-bubbles="top" data-loaded="true">
                                    
                                    <div class="don-share-facebook"><a href="https://www.facebook.com/sharer.php?u={game-url}" target="_blank" class="socialhref_click" title="Share on Facebook">
                                        <span class="don-btn don-btn-facebook don-btn-ico"><i class="fa fa-facebook"></i></span>
                                        <span class="don-count">87</span></a>
                                    </div>
                                    <div class="don-share-twitter"><a href="https://twitter.com/share?url={game-url}" target="_blank" title="Share on Twitter" class="socialhref_click">
                                        <span class="don-btn don-btn-twitter don-btn-ico"><i class="fa fa-twitter"></i></span>
                                        <span class="don-count"> </span></a>
                                    </div>
                                </div>

**Comment by gabu69** [comment]
Sorry, didnt see the reply @diegoquinteiro .

Will @IevaUzk reply be enough? I think playbuzz quizzes are quite important for many of us

**Comment by demoive** [comment]
Yes @gabu69, the provided example should be enough. Thanks.

I've tracked this as a feature request to support the Playbuzz plugin by default.

In the mean time, you could try defining your own custom rules to support this plugin as it will not conflict when a new version of the plugin is released to possibly support this case. To assist in defining your own rules, we've extended our documentation with a section dedicated to help configuring the [Custom Transformer Rules](https://github.com/facebook/facebook-instant-articles-sdk-php/blob/master/docs/QuickStart.md#custom-transformer-rules). Hopefully it provides enough information for you to define new ones or make any adjustments necessary to get the transformation to work in your environment.

If you find you are unable to define your own custom rules to resolve this case (1) we'd appreciate feedback on what part of the documentation that is lacking and (2) the [community on the WP Forum](https://wordpress.org/support/plugin/fb-instant-articles) can occasionally be quite helpful in providing support.

**Comment by gabu69** [comment]
Is there any news with this integration? I cant create the classes since I got now idea how to do it, im just an average publisher with no technical capabilities to do this :S

**Comment by everton-rosario** [comment]
@gabu69 and @levaUzk Hey, sorry for the long wait on this thread.

I'll take a look on this plugin integration. 


Please, can you share here more info about your transformation problem?

We need you to click in the link "**[Toggle debug information]**" in the "Instant Articles" meta-box and then copy and past the both editor's text: "**Source Markup**" and "**Transformed Markup**"

**Comment by juliobranha** [comment]
I wrote some code that will execute the shortcode and then transform the markup to the one required for IAs. Unfortunately, later on the plugin will strip the markup.
If anyone has a solution for that I leave u here the code u can add to functions.php in ex.
`/**
 * playbuzz_instant_articles regex for embeding playbuzz in FB instant articles
 * @param  string $content post content
 * @return string post content modified
 */
function playbuzz_instant_articles( $content ) {
    // if playbuzz plugin
    if ( shortcode_exists( 'playbuzz-item' ) ) {
        do_shortcode( $content );
    }

    if ( strpos( $content, 'pb_feed' ) !== false ) {
        $content = preg_replace( '~(<p>\s*)?(<script.*src="//cdn.playbuzz.com/widget/feed.js".*>.*</script>)(\s*</p>)?~i', '<figure class="op-interactive"><iframe class="no-margin">$2', $content );
        $content = preg_replace( '~(<p>\s*)?(<div.*class="pb_feed".*>.*</div>)(\s*</p>)?~i', '$2</iframe></figure>', $content);
    }

    return $content;
}

add_filter( 'instant_articles_content', 'playbuzz_instant_articles' );`

**Comment by everton-rosario** [comment]
@juliobranha I like your approach of find/replace. 
Because it doesn't use the transformer, it doesn't handle correctly the transformation warnings. I'll have a PR soon, hoping you can help me test as well.

**Comment by juliobranha** [comment]
@everton-rosario Sure :) more than happy to do that, still looking and how to avoid the markup getting stripped after applying the filter

**Comment by juliobranha** [comment]
taking a look to transformation rules now, didn't know much about :), maybe then only de do_shortcode part will be needed on the filter

**Comment by juliobranha** [comment]
For now adding a couple of rules and the code above seem to work, but would look to implement a proper transformer rule and do it in the proper way :)
```
{
    "rules":
    [{
        "class": "PassThroughRule",
        "selector": "figure"
    }, {
    "class": "PassThroughRule",
        "selector": ".pb_feed"
    }]
}
```

**Comment by juliobranha** [comment]
BTW, I think ` do_shortcode( $content );` should be a part of the plugin core unless there is a reason against that

**Comment by everton-rosario** [comment]
@juliobranha Check this PR #417 

To test, make sure you are on latest version of SDK and the WP Plugin.
[Instructions on how to do this setup](https://github.com/Automattic/facebook-instant-articles-wp#development-environment).


Please join our group @ Facebook: https://web.facebook.com/groups/629655867187397/
There we have releases, announcements, events and all exchange between all Open Source tools regarding Instant Articles.

**Comment by everton-rosario** [comment]
Closing this. If you find any issue with this implementation, please reopen.

**Comment by gabu69** [comment]
Got quite confused, its still not working right?

**Comment by everton-rosario** [comment]
This is fixed.
You gotta use the latest version of Facebook Instant Articles + Latest version of Playbuzz plugin, @gabu69.

If you see something odd, please file a new issue with repro steps, please.
